src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -163,13 +168,14 @@ public DType(DKind kind)
             Metadata = null;
             _attachmentType = null;
             AssociatedDataSources = new HashSet<IExternalTabularDataSource>();
+            DisplayNameProviders = new HashSet<BidirectionalDictionary<string, string>>();
```

Comment:
>DisplayNameProviders = new HashSet<BidirectionalDictionary<string, string>>(); [](http://example.com/codeflow?start=12&length=78)

Is there a penalty to adding this to every dtype? (ie, should it be null most of the time)

Reply:
Or share a single static readonly instance?

Reply:
It's just another empty HashSet, it shouldn't be a problem